### PR TITLE
LPS-185207 Make className a partitioned control table

### PR DESCRIPTION
**Pull request by achaparro**
Hi @shuyangzhou,

This is the pull we were talking about yesterday in the Core Infra call.

We agreed to make these changes in this slack thread some time ago:
https://liferay.slack.com/archives/C04E68K19H9

I have removed the iteration in the ClassNameLocalServiceImpl as you requested yesterday.

This a draft version, we need to still do some testing. Let me know what you think.

Thank you.

cc/ @luisortiz89

### CODE REVIEW

File: portal-impl/src/com/liferay/portal/service/impl/ClassNameLocalServiceImpl.java [hunk 1]
```diff
@@ -146,11 +152,23 @@ public void invalidate() {
...
+
 	private static final Log _log = LogFactoryUtil.getLog(
 		ClassNameLocalServiceImpl.class);
 
 	private static final Map<String, ClassName> _classNames =
 		new ConcurrentHashMap<>();
 	private static final ClassName _nullClassName = new ClassNameImpl();
 
+	@BeanReference(type = CompanyLocalService.class)
+	private CompanyLocalService _companyLocalService;
```
Comment (shuyangzhou) [inline portal-impl/src/com/liferay/portal/service/impl/ClassNameLocalServiceImpl.java:172]: @achaparro what's this unused reference for?

**Comment by shuyangzhou** [comment]
Overall it looks fine, once you finalize the whole thing with testing, I can do another round of review for it.

File: portal-impl/src/com/liferay/portal/service/impl/ClassNameLocalServiceImpl.java (hunk 1 above)
Comment (achaparro) [reply portal-impl/src/com/liferay/portal/service/impl/ClassNameLocalServiceImpl.java:172]: hey @shuyangzhou, I have just forgot to remove that reference when we removed the company iteration from the service. I will delete it in the next pull request

**Comment by achaparro** [comment]
Thanks @shuyangzhou, I will verify everything works as expected and send the final pull request to you.

**Pull request closed without merging**